app/src/org/commcare/views/DrawView.java

```diff
@@ -0,0 +1,158 @@
+    private final Path mCurrentPath;
+    private final Paint mBitmapPaint;
+    private File mBackgroundBitmapFile;
+    private final Paint paint;
+    private final Paint pointPaint;
+    private float mX, mY;
+    private Callback callback;
+
+    public DrawView(final Context c, Paint paint, Paint pointPaint) {
+        super(c);
+
+        this.paint = paint;
+        this.pointPaint = pointPaint;
+        isSignature = false;
+        mBitmapPaint = new Paint(Paint.DITHER_FLAG);
+        mCurrentPath = new Path();
+        setBackgroundColor(0xFFFFFFFF);
+        mBackgroundBitmapFile = SignatureWidget.getTempFileForDrawingCapture();
+    }
+
+    public DrawView(Context c, boolean isSignature, File f, Paint paint, Paint pointPaint) {
+        this(c, paint, pointPaint);
+
+        this.isSignature = isSignature;
+        mBackgroundBitmapFile = f;
+    }
+
+    public void setCallback(Callback callback) {
+        this.callback = callback;
+    }
+
+    public void removeCallback() {
+        this.callback = null;
+    }
+
+    public void reset() {
+        Display display = ((WindowManager)getContext().getSystemService(
+                Context.WINDOW_SERVICE)).getDefaultDisplay();
+        int screenWidth = display.getWidth();
+        int screenHeight = display.getHeight();
+        resetImage(screenWidth, screenHeight);
+    }
+
+    public void resetImage(int w, int h) {
+        if (mBackgroundBitmapFile.exists()) {
+            mBitmap = MediaUtil.getBitmapScaledToContainer(
+                    mBackgroundBitmapFile, w, h).copy(
+                    Bitmap.Config.RGB_565, true);
+            // mBitmap =
+            // Bitmap.createScaledBitmap(BitmapFactory.decodeFile(mBackgroundBitmapFile.getPath()),
+            // w, h, true);
+            mCanvas = new Canvas(mBitmap);
+        } else {
+            mBitmap = Bitmap.createBitmap(w, h, Bitmap.Config.RGB_565);
+            mCanvas = new Canvas(mBitmap);
+            mCanvas.drawColor(0xFFFFFFFF);
+            if (isSignature) {
+                drawSignLine();
+            }
+        }
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        super.onSizeChanged(w, h, oldw, oldh);
+        resetImage(w, h);
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        canvas.drawColor(getResources().getColor(R.color.grey));
+        canvas.drawBitmap(mBitmap, 0, 0, mBitmapPaint);
+        canvas.drawPath(mCurrentPath, paint);
+    }
+
+    private void touch_start(float x, float y) {
+        mCurrentPath.reset();
+        mCurrentPath.moveTo(x, y);
+        mX = x;
+        mY = y;
+    }
+
+    public void drawSignLine() {
+        mCanvas.drawLine(0, (int)(mCanvas.getHeight() * .7),
+                mCanvas.getWidth(), (int)(mCanvas.getHeight() * .7), paint);
+    }
+
+    private void touch_move(float x, float y) {
+        double distance = Math.sqrt(Math.pow(x - mX, 2) + Math.pow(y - mY, 2));
```

Comment:
This change will make sure that the save button is enabled only when user has drawn something with a length of 5 at least.

Reply:
is this the only change in `DrawView` ?

Reply:
No, I meant that this is the logic which decides whether or not we should enable draw button. 
I think I should move the refractoring to a separate PR. It'll be simpler since the commits are separate and I can then change the base of this PR as well.